hedera-node/src/main/java/com/hedera/services/state/initialization/HfsSystemFilesManager.java

```diff
@@ -235,48 +246,94 @@ private void materialize(FileID fid, HFileMeta info, byte[] contents) {
+			String defaultResource,
 			Consumer<ServicesConfigurationList> onSuccess
 	) {
-		var disFid = fileNumbers.toFid(disNum);
-		if (!hfs.exists(disFid)) {
+		final var sysFileFid = fileNumbers.toFid(sysFileNum);
+		if (!hfs.exists(sysFileFid)) {
 			bootstrapInto(
-					disFid,
-					resource,
+					sysFileFid,
+					sysFileName,
 					() -> asSerializedConfig(
-							resource,
-							properties.getStringProperty(jutilLocProp)));
+							sysFileName,
+							properties.getStringProperty(externalLocProp),
+							defaultResource,
+							errorLogIfAnyForFailureToLoad(sysFileName)));
 		}
-		var config = loadFrom(disFid, resource, ServicesConfigurationList::parseFrom);
+		final var config = loadFrom(sysFileFid, sysFileName, ServicesConfigurationList::parseFrom);
 		onSuccess.accept(config);
 	}
 
-	private byte[] asSerializedConfig(String resource, String propsLoc) throws IOException {
-		try (InputStream fin = Files.newInputStream(Paths.get(propsLoc))) {
-			var jutilProps = new Properties();
-			jutilProps.load(fin);
-			var config = ServicesConfigurationList.newBuilder();
-			var sb = new StringBuilder(String.format("Bootstrapping network %s from '%s':", resource, propsLoc));
-			jutilProps.entrySet()
-					.stream()
-					.sorted(Comparator.comparing(entry -> String.valueOf(entry.getKey())))
-					.peek(entry -> sb.append(String.format(
-							"\n  %s=%s",
-							String.valueOf(entry.getKey()),
-							String.valueOf(entry.getValue()))))
-					.forEach(entry ->
-							config.addNameValue(Setting.newBuilder()
-									.setName(String.valueOf(entry.getKey()))
-									.setValue(String.valueOf(entry.getValue()))));
+	private String errorLogIfAnyForFailureToLoad(String sysFileName) {
+		return PERMISSIONS_SYS_FILE_NAME.equals(sysFileName)
+				? "Could not bootstrap permissions, only superusers will be able to perform HAPI operations!"
+				: "Could not bootstrap properties, likely benign but resources should be double-checked!";
+	}
+
+	private byte[] asSerializedConfig(
+			String sysFileName,
+			String externalPropsLoc,
+			String defaultResource,
+			String errorLog
+	) {
+		final var externalSrcMsg = String.format("Bootstrapping %s from '%s':", sysFileName, externalPropsLoc);
+		final var resourceSrcMsg = String.format("Bootstrapping %s from resource '%s':", sysFileName, defaultResource);
+
+		return configBytesFrom(
+				() -> Files.newInputStream(Paths.get(externalPropsLoc)),
+				externalSrcMsg
+		)
+				.or(() ->
+						configBytesFrom(
+								() -> {
+									var in = HfsSystemFilesManager.class.getClassLoader()
+											.getResourceAsStream(defaultResource);
+									if (in == null) {
+										throw new IOException("Could not load resource '" + defaultResource + "'");
+									}
+									return in;
+								},
+								resourceSrcMsg
+						)
+				).or(() -> {
+							log.error(errorLog);
+							return Optional.of(ServicesConfigurationList.getDefaultInstance().toByteArray());
+						}
+				).get();
+	}
```

Comment:
Seems to me the logic here is to take properties from external property file first. If the external file doesn't exist, then take it from property file in `resource` location. Again, if it doesn't exist, take from grpc channel. 
The priorities of these sources are fine to me. My question is will this force the user to have re-write all properties in each property source? In another word, if we first get the properties from grpc, then from property file in `resource`, then from external property file. That way, the use only need to change one property, he or she only needs to put that property in external property file. Eventually that property value will have the final say. Any other properties will take their values from `resource` property file, or from grpc, or from default.

Reply:
Hmm, if there is already a saved state, this method will never be called.

Only if there is no file `0.0.121` or `0.0.122` in state will this method be called.

Reply:
Ohh, I see. I was confused and thought the `loadConfigWithJutilPropsFallback(...)` method is used to implement configuration call backs as well. Thx for the clarification.